app/jobs/signal_adapter/attach_contributors_avatar.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module SignalAdapter
+  class AttachContributorsAvatar < ApplicationJob
+    queue_as :attach_signal_avatar
+
+    def perform(contributor)
+      avatar = "/app/signal-cli-config/avatars/profile-#{contributor.signal_phone_number}"
+      return unless File.file?(avatar)
+
+      contributor.avatar.attach(io: File.open(avatar), filename: "#{contributor.name.delete(' ').underscore}_profile_pic")
```

Comment:
Are you sure `name.delete(' ').underscore` is enough to ensure we always get valid file names. Although I don’t think that ActiveStorage uses the filename for the actual files written to disk, having invalid file names might cause unexpected problems when switching the storage service or downloading files.

You could use this method from ActiveStorage: https://api.rubyonrails.org/classes/ActiveStorage/Filename.html#method-i-sanitized

Reply:
I am not sure that there is a universally accepted "valid" filename. For example, I saw a lot of people dealing with issues when there are umlauts as part of the filename. See [Rails 6: Files with special characters in the name crash app when accessing amazon S3 with active storage](https://stackoverflow.com/questions/60223440/rails-6-files-with-special-characters-in-the-name-crash-app-when-accessing-amaz), for example.

But looking at the `ActiveStorage.new(filename).sanitized` method you linked to, you might be more interested in characters like `$`, in which case I think it would be helpful to sanitize them as you suggested since we don't validate our first and last names so we cannot be sure that they will not contain any "illegal" characters.

I also used [ActiveSupport::Inflector.transliterate](https://apidock.com/rails/v6.0.0/ActiveSupport/Inflector/transliterate) to address special characters. Maybe this is overkill, or maybe we shouldn't try to use the contributor's name at all to save as a filename.

I considered using `File.basename(avatar)`, but this would include a `+` character, which is considered a [Characters to Avoid in Filenames and Directories](https://www.mtu.edu/umc/services/websites/writing/characters-avoid/) so then we would have to call `.phony_formatted(spaces: '-')`

Don't know which is better, to be honest. I don't really care, except trying to avoid as many issues as possible for our future selves.

Reply:
How about just using the contributor ID as the filename?

Reply:
Yeah, that's much simpler hahaha ... Here I was making it more complicated than it need be. Thanks :blush: